core/bugnote_api.php

```diff
@@ -753,10 +753,38 @@ function bugnote_clear_cache( $p_bugnote_id = null ) {
 
 	if( null === $p_bugnote_id ) {
 		$g_cache_bugnote = array();
+		$g_cache_bugnotes = array();
 	} else {
-		unset( $g_cache_bugnote[(int)$p_bugnote_id] );
+		if( isset( $g_cache_bugnote[(int)$p_bugnote_id] ) ) {
+			$t_note_obj = $g_cache_bugnote[(int)$p_bugnote_id];
+			bugnote_clear_bug_cache( $t_note_obj->bug_id );
+			unset( $g_cache_bugnote[(int)$p_bugnote_id] );
```

Comment:
My question is whether bugnote_clear_bug_cache() already clears the note from the $g_cache_bugnote, and then you are unsetting it again.  It feels to me that the unset line is redundant.